From a12caf730b1985f21eb5c689b4eea0cab01c5f25 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 23 Sep 2023 14:10:29 +0200
Subject: [PATCH] openmp: add 5.1/5.2, fixes openmp with llvm v17

llvm v17 defaults to 5.1 and without this meson fails to find
openmp: 'ERROR: Dependency "openmp" not found, tried system'

Add 5.2 as well while at it.
---
 mesonbuild/dependencies/misc.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mesonbuild/dependencies/misc.py b/mesonbuild/dependencies/misc.py
index b186b5dda83d..b41f3c2610d7 100644
--- a/mesonbuild/dependencies/misc.py
+++ b/mesonbuild/dependencies/misc.py
@@ -85,6 +85,8 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
 class OpenMPDependency(SystemDependency):
     # Map date of specification release (which is the macro value) to a version.
     VERSIONS = {
+        '202111': '5.2',
+        '202011': '5.1',
         '201811': '5.0',
         '201611': '5.0-revision1',  # This is supported by ICC 19.x
         '201511': '4.5',
